service/routes.py

```diff
@@ -594,43 +449,59 @@ def get(self, shopcart_id, item_id):
     @api.marshal_with(item_model)
     def put(self, shopcart_id, item_id):
         """
-        Update an Item
+        Update a Item
 
-        This endpoint will update the Item based on the posted body according to the shopcart_id and item_id specified
-        in the path.
+        This endpoint returns just a item
         """
         check_content_type(DEFAULT_CONTENT_TYPE)
-
         app.logger.info("Request update item with shopcart_id: %s and item_id: %s", shopcart_id, item_id)
         shopcart = Shopcart.get_by_id(shopcart_id)
         if not shopcart:
             abort(
                 status.HTTP_404_NOT_FOUND,
                 f"Shopcart with id '{shopcart_id}' could not be found."
             )
-
         item = Item.get_by_id(item_id)
         if not item:
             abort(
                 status.HTTP_404_NOT_FOUND,
                 f"Item with id '{item_id}' could not be found."
             )
-        item.deserialize(api.payload)
+        data = api.payload
 
-        if item.quantity <= 0:
-            app.logger.error("Invalid item quantity assignment to %s.", item.quantity)
+        if data["name"] is None:
+            app.logger.info("Missing information about item name")
+            abort(
+                status.HTTP_400_BAD_REQUEST,
+                "Must enter the name of item."
+            )
+        if data["quantity"] is None:
+            app.logger.info("Missing information about item quantity")
+            abort(
+                status.HTTP_400_BAD_REQUEST,
+                "Must enter the quantity of item."
+            )
+        if data["quantity"] <= 0:
+            app.logger.info("Can not update item with given quantity")
             abort(
                 status.HTTP_400_BAD_REQUEST,
                 "Quantity of the item must be positive."
             )
-
-        if item.price < 0:
-            app.logger.error("Invalid item price assignment to %s.", item.price)
+        if data["price"] is None:
+            app.logger.info("Missing information about item price")
+            abort(
+                status.HTTP_400_BAD_REQUEST,
+                "Must enter the price of item."
+            )
+        if data["price"] <= 0:
+            app.logger.info("Can not update item with negative or zero price")
             abort(
                 status.HTTP_400_BAD_REQUEST,
                 "Price of the item must be positive."
             )
-
+        item.deserialize(data)
+        item.id = item_id
+        item.shopcart_id = shopcart_id
```

Comment:
we do not need to assign `shopcart_id` here because it already exists in the object we obtain from Item.get_by_id(item_id).